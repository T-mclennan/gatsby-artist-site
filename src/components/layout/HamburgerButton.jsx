import React from 'react';
import { HamburgerSlider } from 'react-animated-burgers';
import PropTypes from "prop-types";
import * as styles from './Navbar.module.css';

function HamburgerButton({open, toggle}) {

  return (
    <div className={styles.hamburgerContainer}>
      <HamburgerSlider buttonWidth={35} isActive={open} toggleButton={toggle} barColor="#d4380d" />
    </div>
  )
}

HamburgerButton.propTypes = {
  open: PropTypes.bool.isRequired,
  toggle: PropTypes.func.isRequired
}

export default HamburgerButton;
