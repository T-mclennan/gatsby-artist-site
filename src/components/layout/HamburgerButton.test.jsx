import React from "react";
import renderer from "react-test-renderer";
import HamburgerButton from "./HamburgerButton";

describe("HamburgerButton", () => {
  it("Has properly rendered", () => {
    const tree = renderer
      .create(<HamburgerButton />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})