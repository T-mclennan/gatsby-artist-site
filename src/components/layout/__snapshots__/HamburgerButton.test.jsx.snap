// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HamburgerButton Has properly rendered 1`] = `
<div>
  <button
    aria-label="Navigation"
    className="sc-bdvvtL dzuxJW"
    onClick={[Function]}
  >
    <div
      className="sc-gsDKAQ lbFyhs"
    >
      <span
        className="sc-dkPtRN sc-fFeiMQ eZphlh joKDaR"
      />
    </div>
  </button>
</div>
`;
